src/values.rs

```diff
@@ -1429,16 +1445,21 @@ mod test {
         let registry = ProgramRegistry::<CoreType, CoreLibfunc>::new(&program).unwrap();
 
         // Call to_jit to get the value of the enum with tag value out of range
-        let _ = Value::Enum {
+        let result = Value::Enum {
             tag: 2,
             value: Box::new(Value::Uint8(10)),
             debug_name: None,
         }
-        .to_ptr(&Bump::new(), &registry, &program.type_declarations[1].id);
+        .to_ptr(&Bump::new(), &registry, &program.type_declarations[1].id)
+        .unwrap_err();
+
+        let error = result.to_string().clone();
+        let error_msg = error.split("\n").collect::<Vec<&str>>()[0];
+
+        assert_eq!(error_msg, "Variant index out of range.");
     }
 
     #[test]
-    #[should_panic(expected = "An enum without variants cannot be instantiated.")]
```

Comment:
Here you could leave the `should_panic` and change the `unwrap_err` for just `unwrap`. I think it's simpler

Reply:
I think I could not. With this changes, program would not panic. In fact, the test didn't pass because it didn't panic.